airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -699,6 +700,36 @@ class FormStream(Stream):
     created_at_field = "createdAt"
 
 
+class FormSubmissionStream(Stream):
+    """Marketing Forms, API v1
+    This endpoint requires the forms scope.
+    Docs: https://legacydocs.hubspot.com/docs/methods/forms/get-submissions-for-a-form
+    """
+
+    url = "/form-integrations/v1/submissions/forms"
+    limit = 50
+    updated_at_field = "updatedAt"
+
+    def _transform(self, records: Iterable) -> Iterable:
+        for record in super()._transform(records):
+            keys = record.keys()
+            if "id" not in keys:
+                record["id"] = uuid4()
```

Comment:
hey @gergelylendvai why you're creating a `id` for the record?

Reply:
it's not contained by default since the return values from the api look like this:
```
"results": [
    {
      "submittedAt": 1547130102576,
      "values": [
        {
          "name": "email",
          "value": "example@example.com"
        },
        {
          "name": "company",
          "value": "Example Co"
        },
        {
          "name": "phone",
          "value": "123 555 4567"
        }
      ],
      "pageUrl": "https://www.example.com/form"
    }
```
However the results can be paging and the new format is used which I think is implemented to `v3` API:
https://github.com/airbytehq/airbyte/blob/0eba52e002235caab14cdbb8f4c5847f8b8b0930/airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py#L394
```
"paging": {
    "next": {
      "after": "YmYyZjI4MGYtMjRjZS00NjY2LTkzMWItMzFmNTBhZjhlOGUzXzE1NDcwNjk5Mzg0OTQ",
      "link": "?after=YmYyZjI4MGYtMjRjZS00NjY2LTkzMWItMzFmNTBhZjhlOGUzXzE1NDcwNjk5Mzg0OTQ&limit=4"
    }
  }
```
And during the fetching of these records there is a line which fails because there is no `id` returned in the response
https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py#L344

Reply:
All things considered, I've removed the `id` in the end